src/human-readable/types/utils.ts

```diff
@@ -198,43 +218,57 @@ export type SplitParameters<
   : []
 type Pop<T extends readonly number[]> = T extends [...infer R, any] ? R : []
 
-export type _ValidateAbiParameter<TAbiParameter extends AbiParameter> =
-  // Validate `name`
-  (
-    TAbiParameter extends { name: string }
-      ? ValidateName<TAbiParameter['name']> extends infer Name
-        ? Name extends TAbiParameter['name']
-          ? TAbiParameter
-          : // Add `Error` as `name`
-            Merge<TAbiParameter, { readonly name: Name }>
-        : never
-      : TAbiParameter
-  ) extends infer Parameter
-    ? // Validate `type` against `AbiType`
-      (
-        ResolvedConfig['StrictAbiType'] extends true
-          ? Parameter extends { type: AbiType }
-            ? Parameter
-            : Merge<
-                Parameter,
-                {
-                  readonly type: Error<`Type "${Parameter extends {
-                    type: string
-                  }
-                    ? Parameter['type']
-                    : string}" is not a valid ABI type.`>
-                }
-              >
-          : Parameter
-      ) extends infer Parameter2 extends { type: unknown }
-      ? // Convert `(u)int` to `(u)int256`
-        Parameter2['type'] extends `${infer Prefix extends
-          | 'u'
-          | ''}int${infer Suffix extends `[${string}]` | ''}`
-        ? Merge<Parameter2, { readonly type: `${Prefix}int256${Suffix}` }>
-        : Parameter2
+export type ValidateType<
+  TType extends string,
+  Strict extends boolean = ResolvedConfig['Strict'],
+> = Strict extends true ? (TType extends AbiType ? true : false) : true
```

Comment:
This was added so that when parsing struct properties we wont have a `invalid type` error.